### ISSUE TITLE
Median display

### ISSUE THREAD

**Issue by voidref** [treyp/thebutton#3]
Have you considered adding the median button click to the data? It's much more indicative of what the typical click time is than average.

I submitted a pull against the original monitor for this, but he seems to have abandoned the project already.

**Comment by treyp** [comment]
Hmm, this wouldn't _that_ difficult considering the histogram data structure is already built. I might be able to get this out tonight.

**Comment by voidref** [comment]
In my implementation for the other version, I just sorted a list of times, and picked the middle one (rounded down). It was pretty simplistic, and probably not 100% accurate given some of items were multi-click, but I think it was 'close enough' to give a meaningful value.

Thanks!

**Comment by treyp** [comment]
Yep, that's the simplest way and probably the easiest to read. I would leave it at that, but I just want to be careful about efficiency with large data sets (30k+ entries). If we're calculating this every time a new click comes in, we'd end up creating large temporary arrays every few seconds. Instead, I'm working on a short method using the histogram data, which wouldn't require more than a few tracking variables for calculation, and at the worst case would iterate over the 60 integers in that array.

Histogram array is basically an index that looks like this for reference:

``` javascript
histogram[59] = 49; // 49 clicks at 60 seconds
histogram[58] = 85; // 85 clicks at 59 seconds
histogram[57] = 34; // 34 clicks at 58 seconds
histogram[56] = 22; // 22 clicks at 57 seconds
// etc.
```

I should be able to find the median in there fairly efficiently!

**Comment by treyp** [comment]
Closing - implemented in 87081937864bbcdea38e86725398e7313a36fc46

**Comment by voidref** [comment]
Oh man, thanks so much! I would give you a click if I could. =)
